providers/cstor/cstor-disk-pool/test.yml

```diff
@@ -22,12 +22,29 @@
             executable: /bin/bash
           register: node_count
 
-        - name: Get list of disks attached
-          shell: kubectl get disks -l ndm.io/disk-type=disk --no-headers -o custom-columns=:metadata.name
+        - name: Getting Used disks
+          shell: kubectl get sp -o jsonpath='{.items[*].spec.disks.diskList}' | tr " " "\n" |sed 's/[][]//g'
           args:
             executable: /bin/bash
-          register: diskList
-          failed_when: "'disk-' not in diskList.stdout"
+          register: used_disks
+      
+        - local_action: copy content={{ used_disks.stdout }} dest="./used_disks"
+
+        - name: Getting all disks
+          shell: kubectl get disks -o jsonpath='{.items[*].metadata.name}' | tr " " "\n"
+          register: all_disks
+
+        - local_action: copy content={{ all_disks.stdout }} dest="./all_disks"
+
+        - name: Copying content
+          shell: cat all_disks used_disks >> tmp_disks
+
+        - name: Getting list of unused disks attached
+          shell: |
+            sort used_disks > a1
+            sort all_disks > a2   
+            diff a2 a1 | grep -e "<" -e ">" | awk '{print $2}'        
```

Comment:
Not sure if this is a relevant review comment! can we keep all this logic separate somewhere - improved NDM operators should be capable of updating attributes in the disk CR on whether it is clamined (part of pool) or no. In which case, the util can be updated and the test can remain untouched ?

Reply:
@ksatchit SP is going to deprecate in near future as discussed with dev-team so the entire picture of our util for finding the unused disk is going to change. This PR is temporary workaround once we will have proper labels in our disk CRs we will write a separate util for this purpose.

Reply:
Thanks @shashank855